src/rust/plugin-work-queue/src/server.rs

```diff
@@ -0,0 +1,144 @@
+use rust_proto::plugin_work_queue::{
+    plugin_work_queue_service_server::{
+        PluginWorkQueueService,
+        PluginWorkQueueServiceServer,
+    },
+    AcknowledgeRequest,
+    AcknowledgeResponse,
+    GetExecuteAnalyzerRequest,
+    GetExecuteAnalyzerResponse,
+    GetExecuteGeneratorRequest,
+    GetExecuteGeneratorResponse,
+    PutExecuteAnalyzerRequest,
+    PutExecuteAnalyzerResponse,
+    PutExecuteGeneratorRequest,
+    PutExecuteGeneratorResponse,
+    _AcknowledgeRequest,
+    _AcknowledgeResponse,
+    _GetExecuteAnalyzerRequest,
+    _GetExecuteAnalyzerResponse,
+    _GetExecuteGeneratorRequest,
+    _GetExecuteGeneratorResponse,
+    _PutExecuteAnalyzerRequest,
+    _PutExecuteAnalyzerResponse,
+    _PutExecuteGeneratorRequest,
+    _PutExecuteGeneratorResponse,
+};
+use tonic::{
+    transport::Server,
+    Request,
+    Response,
+    Status,
+};
+
+#[derive(Debug, thiserror::Error)]
+pub enum PluginWorkQueueError {}
+
+pub struct PluginWorkQueue {}
+
+impl PluginWorkQueue {
+    async fn _put_execute_generator(
```

Comment:
Can you talk more about this apparent underscore convention that's being used for these method names, as well as for the imported types above? It appears to be some kind of "private implementation" convention (to distinguish between the "raw" Protobuf API and a transport-independent business logic representation). If that's the case, it's a bit confusing, because it's bringing Python conventions into a Rust codebase, and also running up against Rust's use of a leading underscore to denote unused variables. 

For the types, could we do something like naming them `FooProto` instead of `_Foo`? And for the functions, could they be called explicitly, like `PluginWorkQueue::put_execute_generator(<THE_INSTANCE>, request)`?

Reply:
I can use `FooProto` instead of `_Foo`, no problem.

I can see if ucfs is necessary, it may be that Rust can understand how to dispatch based on the argument type.